packages/temporal-time-utils/EntityWorkflow.ts

```diff
@@ -0,0 +1,74 @@
+import * as wf from "@temporalio/workflow";
+import { SignalDefinition, QueryDefinition } from "@temporalio/common";
+
+const noop = async () => {};
+type ThingToInvoke =
+  | { activity: string; activityOptions: wf.ActivityOptions }
+  | { workflow: string; workflowOptions: wf.ChildWorkflowOptions };
+export class Entity<Input = any, Update = any> {
+  MAX_ITERATIONS: number;
+  setup: (input: Input) => Promise<void>;
+  cleanup: () => Promise<void>;
+  thingToInvoke: ThingToInvoke;
```

Comment:
`thingToInvoke` should be of type `(input: Update) => void` instead of limiting it to either an activity or workflow.

I'd rename this to `updateFunction` too.